src/routes/repetition/components/Chart.tsx

```diff
@@ -0,0 +1,68 @@
+import { useFocusEffect } from '@react-navigation/native'
+import React, { ReactElement, useCallback, useEffect, useState } from 'react'
+import { heightPercentageToDP as hp } from 'react-native-responsive-screen'
+import { LineSegment, VictoryAxis, VictoryBar, VictoryChart, VictoryLabel } from 'victory-native'
+
+import theme from '../../../constants/theme'
+import useLoadAsync from '../../../hooks/useLoadAsync'
+import { getNumberOfWordsInEachSection, sections } from '../../../services/RepetitionService'
+import { getLabels } from '../../../services/helpers'
+
+const { chartColor1, chartColor2, chartColor3, chartColor4 } = theme.colors
+const barColors = [chartColor1, chartColor2, chartColor2, chartColor2, chartColor2, chartColor2, chartColor3]
+const Chart: () => ReactElement = () => {
+  const { new_, learned, sectionOneToFive } = getLabels().repetition.chart
+  const { words } = getLabels().general
+
+  const [chartData, setChartData] = useState<{ y: number; x: number; fill: string }[]>(
+    Array(sections.length).fill({
+      y: 0,
+      fill: chartColor2,
+    })
+  )
+
+  const { data, refresh } = useLoadAsync(getNumberOfWordsInEachSection, undefined)
+
+  useEffect(() => {
+    if (data) {
+      setChartData(data.map((item, index) => ({ y: item, x: index, fill: barColors[index] })))
+    }
+  }, [data])
```

Comment:
I think this could be improved and simplified without the using of a `useState`, `useLoadAsync` and a `useEffect`:

```suggestion
  const getChartData = useCallback(async () => {
    const data = await getNumberOfWordsInEachSection()
    return data.map((item, index) => ({ y: item, x: index, fill: barColors[index] }))
  }, [])
  const { data, refresh } = useLoadAsync(getChartData, undefined)
  const chartData =
    data ??
    Array(sections.length).fill({
      y: 0,
      fill: chartColor2,
    })

```

Reply:
Good idea :+1: